clients/client-py/test/test_client.py

```diff
@@ -18,281 +17,306 @@
-        self.assertEqual(payload, {'A': 123})
-        self.assertEqual(query, {'B': 456})
-
-    def test_calling_convention_3_without_payload_with_query(self):
-        params, payload, query, _, _ = self.client._processArgs(
-            {'args': ['k1', 'k2'], 'name': 'test'},
-            params={'k1': 1, 'k2': 2},
-            query={'B': 456}
-        )
-        self.assertEqual(params, {'k1': 1, 'k2': 2})
-        self.assertEqual(payload, None)
-        self.assertEqual(query, {'B': 456})
-
-    def test_calling_convention_3_with_positional_arguments_with_payload_with_query(self):
-        params, payload, query, _, _ = self.client._processArgs(
+import pytest
+
+pytestmark = [
+    pytest.mark.skipif(os.environ.get("NO_TESTS_OVER_WIRE"), reason="Skipping tests over wire")
+]
+
+REAL_TIME_SLEEP = time.sleep
+
+@pytest.yield_fixture(scope='function')
+def apiRef():
+    subject.config['credentials'] = {
+        'clientId': 'clientId',
+        'accessToken': 'accessToken',
+    }
+    keys = [
+        base.createTopicExchangeKey('primary_key', constant='primary'),
+        base.createTopicExchangeKey('norm1'),
+        base.createTopicExchangeKey('norm2'),
+        base.createTopicExchangeKey('norm3'),
+        base.createTopicExchangeKey('multi_key', multipleWords=True),
+    ]
+    topicEntry = base.createApiEntryTopicExchange('topicName', 'topicExchange', routingKey=keys)
+    entries = [
+        base.createApiEntryFunction('no_args_no_input', 0, False),
+        base.createApiEntryFunction('two_args_no_input', 2, False),
+        base.createApiEntryFunction('no_args_with_input', 0, True),
+        base.createApiEntryFunction('two_args_with_input', 2, True),
+        base.createApiEntryFunction('NEVER_CALL_ME', 0, False),
+        topicEntry
+    ]
+    apiRef = base.createApiRef(entries=entries)
+    clientClass = subject.createApiClient('testApi', apiRef)
+    client = clientClass({'rootUrl': base.TEST_ROOT_URL})
+    # Patch time.sleep so that we don't delay tests
+    sleepPatcher = mock.patch('time.sleep')
+    sleepSleep = sleepPatcher.start()
+    sleepSleep.return_value = None
+    patcher = mock.patch.object(client, 'NEVER_CALL_ME')
+    never_call = patcher.start()
+    never_call.side_effect = AssertionError
+    fakeResponse = ''
+
+    def fakeSite(url, request):
+        gotUrl = urllib.parse.urlunsplit(url)
+        gotRequest = request
+        return fakeResponse
+    fakeSite = fakeSite
+
+    yield apiRef
+
+@pytest.yield_fixture(scope='function')
+def clientClass(apiRef):
+    clientClass = subject.createApiClient('testApi', apiRef)
+    yield clientClass
+
+@pytest.yield_fixture(scope='function')
+def client(clientClass):
+    client = clientClass({'rootUrl': base.TEST_ROOT_URL})
+    yield client
+
+@pytest.yield_fixture(scope='function')
+def patcher(client):
+    patcher = mock.patch.object(client, 'NEVER_CALL_ME')
+    yield patcher
+
+
+def tearDown():
+    time.sleep = REAL_TIME_SLEEP
```

Comment:
This function doesn't do anything. This belongs in the cleanup section of a yield fixture, or we can use fixtures that clean up after themselves like `mocker`.